### PR TITLE
Fixing unescaped parameters when running

### PR DESCRIPTION
**Pull request by WorldSEnder**
Compare in the current version
```
aurman -Qi --cachedir "/tmp/with spaces" aurman-git
#expected to be equivalent to: pacman -Qi --cachedir "/tmp/with spaces" aurman-git
#actually has the effect of   : pacman -Qi --cachedir "/tmp/with" "spaces" aurman-git
```

This pull request should fix this an similar issues by calling `subprocess.run` with an array instead of a single `string`. Fixup of #36, now should pass the travis tests.

### CODE REVIEW

**Comment by polygamma** [comment]
It really is overkill. see: https://github.com/polygamma/aurman/commit/d8323d4cad0557316e470cb3e72a7884816cb56f and https://github.com/polygamma/aurman/commit/b52676e684e38ccca2f35b1c33acc337a173f2c3

**Comment by WorldSEnder** [comment]
It does a bit more by not running things in a shell (would possibly invoke shell expansions a second time). Using `shell=True` is something I'd get fully get rid off where possible.

It also automatically escapes parameters without such manual intervention as in b52676e684e38ccca2f35b1c33acc337a173f2c3 (the cleaner method would be `shlex.quote` imo).

Currently wondering why the build failed here, it worked here: https://travis-ci.org/WorldSEnder/aurman/builds/356047874

**Comment by polygamma** [comment]
I do not really see a problem with using `shell=True` here. It's the user who calls aurman on his own pc with the parameters he wants. The space problem has been fixed, and everything else is nothing which I want to change.

**Comment by WorldSEnder** [comment]
It's not that it would break stuff if you didn't merge this. It's that the proposed solution is more stable.

Consider queries with special characters in them: `aurman -Ss "'startup"`. Not a common case for sure. But manually escaping stuff leads to missing those edge cases.

**Comment by AladW** [comment]
Well, `pkgname` can only contain alphanumerics or `@ . _ + -` anyway, but I guess full paths (e.g. to `cachedir`) can contain arbitrary characters...

**Comment by WorldSEnder** [comment]
Addressing the error in the build: `Error with: 'aurmansolver -S ros-indigo-desktop-full --deep_search'` I can't say for sure what causes that. It works locally on in the travis build on my branch.

**Comment by polygamma** [comment]
I am going to take a closer look tomorrow

**Comment by polygamma** [comment]
First things first: You are right, that this approach is more stable. The problem: It touches too much of the sourcecode, and I am not sure if there may be side effects, hence I won't accept this PR. I'd definitely consider this in a complete rewrite of aurman, but for now it really does not give any benefit. Normally you only yield packages names on the command line, they must not contain "problematic" characters per definition. Now there may be paths for e.g. '--cachedir', only spaces are being handled right now, hence there could still be problematic characters. But that case is not enough for me, to change that much of code. One could just use a sane path, and everything is fine.

tl;dr: This is a thing for a complete rewrite, but nothing for now.

**Comment by polygamma** [comment]
see: https://github.com/polygamma/aurman/compare/5a4093585ee7a576ea2b594917ee2758a1c5252e...32e2ac63a2f91e6d139069418c923612f3a6f3ab
